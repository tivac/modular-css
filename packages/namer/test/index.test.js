"use strict";

var namer = require("../");

describe("modular-css-namer", function() {
    var fn;
    
    beforeEach(() => (fn = namer()));
    
    it("should hash its arguments", function() {
        expect(fn("./fooga", ".fooga")).toMatchSnapshot();
    });
    
    it("should differ within files", function() {
        expect(fn("./fooga", ".fooga")).toMatchSnapshot();
        expect(fn("./fooga", ".booga")).toMatchSnapshot();
    });
    
    it("should re-use selectors for identical inputs", function() {
        expect(fn("./fooga", ".fooga")).toMatchSnapshot();
        expect(fn("./fooga", ".fooga")).toMatchSnapshot();
    });

    it("should differ between files", function() {
        expect(fn("./fooga", ".fooga")).toMatchSnapshot();
        expect(fn("./booga", ".fooga")).toMatchSnapshot();
    });
    
    it("should wrap as necessary", function() {
        var x, y;
        
        for(x = 0; x < 100; x++) {
            for(y = 0; y < 100; y++) {
                fn(`./fooga${x}`, `.fooga${y}`);
            }
        }
        
        expect(fn(`./fooga${15}`, `.fooga${5}`)).toMatchSnapshot();
        expect(fn(`./fooga${1}`, `.fooga${5}`)).toMatchSnapshot();
        expect(fn(`./fooga${49}`, `.fooga${33}`)).toMatchSnapshot();
        expect(fn(`./fooga${55}`, `.fooga${63}`)).toMatchSnapshot();
        expect(fn(`./fooga${26}`, `.fooga${0}`)).toMatchSnapshot();
    });
});
