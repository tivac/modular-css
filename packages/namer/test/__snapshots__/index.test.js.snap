// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modular-css-namer should differ between files 1`] = `"AA"`;

exports[`modular-css-namer should differ between files 2`] = `"BA"`;

exports[`modular-css-namer should differ within files 1`] = `"AA"`;

exports[`modular-css-namer should differ within files 2`] = `"AB"`;

exports[`modular-css-namer should hash its arguments 1`] = `"AA"`;

exports[`modular-css-namer should re-use selectors for identical inputs 1`] = `"AA"`;

exports[`modular-css-namer should re-use selectors for identical inputs 2`] = `"AA"`;

exports[`modular-css-namer should wrap as necessary 1`] = `"PF"`;

exports[`modular-css-namer should wrap as necessary 2`] = `"BF"`;

exports[`modular-css-namer should wrap as necessary 3`] = `"xh"`;

exports[`modular-css-namer should wrap as necessary 4`] = `"zD9B"`;

exports[`modular-css-namer should wrap as necessary 5`] = `"aA"`;
